util/logs/workflow-logger.go

```diff
@@ -111,15 +132,24 @@ func WorkflowLogs(ctx context.Context, wfClient versioned.Interface, kubeClient
 					logCtx.Error(err)
 					return
 				}
+
 				scanner := bufio.NewScanner(stream)
+				//give it more space for long line
+				scanner.Buffer(make([]byte, startBufSize), maxTokenLength)
```

Comment:
I don’t really understand this PR. Is is based on any docs or blog post you could share please?

Reply:
`scanLinesOrGiveLong` is to let scanner avoid `bufio.ErrTooLong` error, which will cause `scanner.Scan()` return false, thus break the loop early, leaving logs in the stream not scanned.  This method is borrowed from [here](https://gist.github.com/AlbinoDrought/c292adc185ea6593338fb5b56ac89aa7). 

Actually, the method above alone is sufficient to solve the issue #9675, but it will be nicer with the complementary of the following.

`scanner.Buffer(make([]byte, startBufSize), maxTokenLength)` is to enlarge the scanner's inner buffer(the default size is 4096), and replace the MaxScanTokenSize(the default value is 64 * 1024) with a larger one.  According to our observation, it is common that users print visual progress bar to stdout. Most of the visual progress bar use the '\r' character to return to the start point of the same line, as to give user-friendly visual effect on terminal. Thus, it forms a very long line for scanner, as scanner treats '\n' as the end of each line by default.  if we don't set more larger MaxScanTokenSize, the progress bar will be chopped in the middle very often. it is not a big deal, but larger one will make the logs looks nicer in this case.

It is worthy to mention that this issue can be solved by another method:
_loop over stream manually without the help of scanner, and get the new line when seeing a new '\n'._  
This method have a drawback, user may print a long line in a very long time, such as the progress-bar-like logs, then the user will wait very long time and then see a very long line log pop up abruptly.